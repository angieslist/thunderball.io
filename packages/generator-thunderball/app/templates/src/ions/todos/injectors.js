import reducers from './reducers';

export default {
  beforeConfigureStore: options => ({
    reducers: {
      ...(options || {}).reducers,
      ...reducers,
    },
  }),
};
