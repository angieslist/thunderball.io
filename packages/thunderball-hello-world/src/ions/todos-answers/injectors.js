import thunk from 'redux-thunk';
import reducers from './reducers';

export default {
  beforeConfigureStore: options => ({
    middleware: [
      ...(options.middleware || []),
      thunk,
    ],
    reducers: {
      ...options.reducers,
      ...reducers,
    },
  }),
};
