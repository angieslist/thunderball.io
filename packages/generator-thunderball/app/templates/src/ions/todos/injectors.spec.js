import injectors from './injectors';

test('is a properly formatted injectors object', () => {
  expect(injectors).toMatchSnapshot();
});

test('beforeConfigureStore registers reducers', () => {
  const options = {};
  expect(injectors.beforeConfigureStore(options)).toMatchSnapshot();
});
