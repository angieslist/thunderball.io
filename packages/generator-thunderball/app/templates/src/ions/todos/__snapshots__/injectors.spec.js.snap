// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`beforeConfigureStore registers reducers 1`] = `
Object {
  "reducers": Object {
    "todos": [Function],
    "visibilityFilter": [Function],
  },
}
`;

exports[`is a properly formatted injectors object 1`] = `
Object {
  "beforeConfigureStore": [Function],
}
`;
